packages/docs/src/content/docs/reference/faq.md

```diff
@@ -0,0 +1,326 @@
+Jest has `setupFilesAfterEnv: "<rootDir>/jest.setup.js"` or a reference may
+point to a file in another workspace in the same monorepo, e.g.
+`setupFiles: ['@org/shared/jest-setup.ts']`. Those entry files may also contain
+imports of internal modules or external dependencies, and so on.
+
+With an easy-to-use plugin API, many plugins have been created by contributors.
+
+## Why is Knip so heavily engineered?
+
+Even though I love the Unix philosophy, at this point I believe for Knip it
+makes sense to have the pieces in a single tool.
+
+Building up the module and dependencies graph requires non-standard module
+resolution and not only static but also dynamic analysis (i.e. actually load and
+execute modules), such as for parsers of plugins to receive the exported value
+of dynamic tooling configuration files. Additionally, [exports consumed by
+external libraries][6] require type information, as supported by the TypeScript
+backend.
+
+The rippling effect of plugins and recursively adding entry files and
+dependencies to build up this graph - as answered with the previous question -
+is also exactly what's meant by ["comprehensive" here][7].
+
+## Does Knip handle Vue or Astro files?
+
+To further increase the coverage of the module graph, non-standard files other
+than JavaScript and TypeScript modules should be included as well. For instance,
+`.mdx` and `.astro` files can import each other, internal modules and external
+dependencies.
+
+Knip includes basic "compilers" for a few common file types (Astro, MDX, Svelte,
+Vue). Knip does not include actual compilers for reasons of potential
+incompatibility with the existing compiler, and dependency size. Knip allows to
+override them with the compiler(s) in your project, and add additional ones for
+other file types.
+
+## Why are the exports of my `.svelte` files not used?
+
+Knip comes with basic "compilers" for a few common non-standard file types.
+They're not actual compilers, they're regular expressions only to extract import
+statements. Override the built-in Svelte "compilers" with the real one in your
+project. Also see the answer to the previous question and [Compilers][8].
+
+## Why isn't production mode the default?
+
+The default mode of Knip includes all source files, tests, dependencies, dev
+dependencies and tooling configuration.
+
+On the other hand, production mode considers only source files and production
+dependencies. Plugins add only production entry files.
+
+Which mode should've been the default? They both have their merits:
+
+- Production mode catches dead production code and dependencies. This mode has
+  the most impact on UX, since less code tends to be faster and safer.
+- Default mode potentially catches more issues, e.g. lots of unused plugins of
+  tooling, including most issues found in production mode. This mode has the
+  most impact on DX, for the same reason.
+
+Also see [production mode][9].
```

Comment:
Were you not thinking of having Knip run both types of analysis all the time? I remember we talked about this at some point.

Reply:
Yes, but doesn't seem feasible other than actually running it twice. We've discussed it here as well: https://github.com/webpro-nl/knip/issues/583#issuecomment-2053981278